pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/relocation/SegmentRelocator.java

```diff
@@ -196,6 +203,10 @@ private void rebalanceTable(String tableNameWithType) {
     rebalanceConfig.setMinimizeDataMovement(_minimizeDataMovement);
 
     try {
+      if (_tablesUndergoingRebalance != null) {
+        LOGGER.debug("Start rebalancing table: {}, adding to tablesUndergoingRebalance", tableNameWithType);
+        _tablesUndergoingRebalance.add(tableNameWithType);
```

Comment:
We can check if the table is successfully added. By the time the thread reaches here, there might be another thread running on the same table

Reply:
done, modified this to have an early return if add returns false